akka-stream/src/main/scala/akka/stream/impl/BoundedSourceQueue.scala

```diff
@@ -86,8 +90,10 @@ import akka.stream.stage.{ GraphStageLogic, GraphStageWithMaterializedValue, Out
                   clearNeedsActivation()
                   run()
                 } // else Queue.isEmpty && setNeedsActivation was true: waiting for next offer
-              } else
+              } else {
                 push(out, next) // and then: wait for pull
+                notifySubscribers() // notify subscribers on having capacity to accept a new element
```

Comment:
It is not optimal, ideally we need to notify subscribers only right after there was luck of free capacity. Trying to notify subscribers when the queue was not previously (immediately) full makes not much sense, but I feel the actual overhead to notify subscribers (if there are no any in the list/queue) is negligible. Do you think we need to introduce a flag to notify subscribers only right after the queue was full?

Reply:
After giving it a bit more thought, I feel we need the flag. It will help to handle a situation when caller subscribes to the notifications and the queue is not full - with the current logic the caller will be blocked until the next `push` call, whereas it should be notified immediately. I'll add this logic soon.